packages/extension-chakra-storefront/src/components/search/index.jsx

```diff
@@ -180,17 +188,36 @@ const Search = (props) => {
         onSearchChange(e)
         shouldOpenPopover()
     }
-
     return (
-        <Box>
-            <Popover isOpen={isOpen} isLazy initialFocusRef={searchInputRef}>
-                <PopoverTrigger>
+        <Box css={styles.searchContainer}>
+            <Popover.Root open={isOpen} isLazy initialFocusEl={() => searchInputRef.current}>
+                <Popover.Trigger asChild>
                     <form onSubmit={onSubmitSearch}>
                         <HStack>
-                            <InputGroup>
-                                <InputLeftElement pointerEvents="none">
-                                    <SearchIcon />
-                                </InputLeftElement>
+                            <InputGroup
+                                startElement={<SearchIcon css={styles.searchIcon} />}
+                                endElement={
+                                    searchQuery.length ? (
+                                        <CloseButton
+                                            size="xs"
+                                            css={styles.clearIcon}
+                                            variant="unstyled"
+                                            aria-label={intl.formatMessage({
+                                                id: 'header.button.assistive_msg.clear_search',
+                                                defaultMessage: 'Clear Search'
+                                            })}
+                                            onClick={() => {
```

Comment:
How about creating a handler function for this event. With the added nesting from chakra v3 components things are starting to look a little messy.